airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -250,8 +259,19 @@ private static Runnable readFromDstRunnable(final AirbyteDestination destination
             throw new DestinationException("Destination process read attempt failed", e);
           }
           if (messageOptional.isPresent()) {
-            LOGGER.info("State in DefaultReplicationWorker from destination: {}", messageOptional.get());
-            messageTracker.acceptFromDestination(messageOptional.get());
+            final AirbyteMessage message = messageOptional.get();
+            LOGGER.info("State in DefaultReplicationWorker from destination: {}", message);
+            messageTracker.acceptFromDestination(message);
+
+            try {
+              if (message.getType() == Type.CONTROL && message.getControl().getType() == AirbyteControlMessage.Type.CONNECTOR_CONFIG) {
+                persistConfigHelper.persistDestinationConfig(jobId, message.getControl().getConnectorConfig().getConfig());
+              }
+            } catch (final ApiException e) {
+              LOGGER.error("Error trying to save updated destination config", e);
+              throw new RuntimeException(e);
+            }
+
```

Comment:
let's this this into private function named along the lines of `updateConnectorConfig` to maintain the same levels of abstraction here.

Reply:
👍🏻 done in [51f1976](https://github.com/airbytehq/airbyte/pull/19811/commits/51f19765184523e0ae88d8ffceaaad6e39a772e1)